src/anndata/experimental/merge.py

```diff
@@ -536,8 +536,6 @@ def concat_on_disk(
     # Argument normalization
     if pairwise:
         raise NotImplementedError("pairwise concatenation not yet implemented")
-    if join != "inner":
-        raise NotImplementedError("only inner join is currently supported")
```

Comment:
I am trying to become more familiar with the code, but it's shocking this was all that was needed.  That being said, the tests do pass and I looked at the results so 🤷

Reply:
I mean, looking at the code path, it makes some sense.  `concat_on_disk` constructs sensible-looking reindexers for the `alt_dim` axis and then passes those on so there shouldn't be a need for the `join` argument anywhere else despite the option to use it